pkg/debug/stop_debug.go

```diff
@@ -0,0 +1,61 @@
+/*
+Copyright 2023 The Rook Authors. All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package debug
+
+import (
+	ctx "context"
+	"fmt"
+	"os"
+	"strings"
+
+	"github.com/rook/kubectl-rook-ceph/pkg/k8sutil"
+	kerrors "k8s.io/apimachinery/pkg/api/errors"
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+func StopDebug(context *k8sutil.Context, clusterNamespace, deploymentName string) {
+
+	err := stopDebug(context, clusterNamespace, deploymentName)
+	if err != nil {
+		fmt.Println(err)
+		os.Exit(1)
+	}
+}
+
+func stopDebug(context *k8sutil.Context, clusterNamespace, deploymentName string) error {
+
+	if !strings.Contains(deploymentName, "-debug") {
+		deploymentName = deploymentName + "-debug"
+	}
+
+	debugDeployment, err := verifyDeploymentExists(context, clusterNamespace, deploymentName)
+	if err != nil {
+		return fmt.Errorf("Missing mon or osd debug deployment name %s. %v\n", deploymentName, err)
+	}
+
+	fmt.Printf("removing debug mode from deployment %s\n", debugDeployment.Name)
+	err = context.Clientset.AppsV1().Deployments(clusterNamespace).Delete(ctx.TODO(), debugDeployment.Name, v1.DeleteOptions{})
+	if err != nil || kerrors.IsNotFound(err) {
```

Comment:
if err != nil && !kerrors.IsNotFound(err) ?

Reply:
actually, it will be `kerrors.IsNotFound(err)` since we do not want to return error if it is not found since we are deleting only

Reply:
if we don not want to return error, only the err is not nil, return err; no need to check whether the err is "NotFound".

Reply:
lets say deletion is complete and we try Delete it will say notFound so to cover that situation I have used that

Reply:
I think @zhucan was correct with this suggestion. We want to return an error in this condition. If it's a notFound error, this error will not be returned.
```suggestion
	if err != nil && !kerrors.IsNotFound(err) {
```

Reply:
done